core/math/vector2.h

```diff
@@ -243,12 +240,11 @@ _FORCE_INLINE_ Vector2 Vector2::operator-() const {
 	return Vector2(-x, -y);
 }
 
-_FORCE_INLINE_ bool Vector2::operator==(const Vector2 &p_vec2) const {
-	return x == p_vec2.x && y == p_vec2.y;
-}
-
-_FORCE_INLINE_ bool Vector2::operator!=(const Vector2 &p_vec2) const {
-	return x != p_vec2.x || y != p_vec2.y;
+std::partial_ordering Vector2::operator<=>(const Vector2 &p_other) const {
+	if (x <=> p_other.x == 0) {
+		return y <=> p_other.y;
+	}
+	return x <=> p_other.x;
```

Comment:
I am a bit concerned on the new operator. I get it, but I am not sure it is an actual improvement readability-wise.

Reply:
It *does* take some time to get used to, I'll give you that. But the readability improvement comes more from consolidation of existing operators, which is more easily demonstrated on larger examples like Vector4:
<details>
<summary>Old</summary>

```cpp
bool Vector4::operator<(const Vector4 &p_v) const {
	if (x == p_v.x) {
		if (y == p_v.y) {
			if (z == p_v.z) {
				return w < p_v.w;
			}
			return z < p_v.z;
		}
		return y < p_v.y;
	}
	return x < p_v.x;
}

bool Vector4::operator>(const Vector4 &p_v) const {
	if (x == p_v.x) {
		if (y == p_v.y) {
			if (z == p_v.z) {
				return w > p_v.w;
			}
			return z > p_v.z;
		}
		return y > p_v.y;
	}
	return x > p_v.x;
}

bool Vector4::operator<=(const Vector4 &p_v) const {
	if (x == p_v.x) {
		if (y == p_v.y) {
			if (z == p_v.z) {
				return w <= p_v.w;
			}
			return z < p_v.z;
		}
		return y < p_v.y;
	}
	return x < p_v.x;
}

bool Vector4::operator>=(const Vector4 &p_v) const {
	if (x == p_v.x) {
		if (y == p_v.y) {
			if (z == p_v.z) {
				return w >= p_v.w;
			}
			return z > p_v.z;
		}
		return y > p_v.y;
	}
	return x > p_v.x;
}
```
</details>
<details>
<summary>New</summary>

```cpp
std::partial_ordering Vector4::operator<=>(const Vector4 &p_other) const {
	if (x <=> p_other.x == 0) {
		if (y <=> p_other.y == 0) {
			if (z <=> p_other.z == 0) {
				return w <=> p_other.w;
			}
			return z <=> p_other.z;
		}
		return y <=> p_other.y;
	}
	return x <=> p_other.x;
}
```
</details>

Another thing somewhat hurting readability is that this new operator is frequently used with a return type of `auto`, which we don't allow. I'm not advocating for its inclusion (the extra verbosity is nice for custom implementations), just noting that it's a contributing factor. It *is* dumb that these new types aren't just part of the language, but seeing as stuff like `int32_t` technically isn't built in either, wacky types are something C++ just has to live with.